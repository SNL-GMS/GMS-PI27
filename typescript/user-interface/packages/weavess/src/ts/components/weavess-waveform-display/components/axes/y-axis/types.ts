export interface YAxisProps {
  /** The height in percentage */
  heightInPercentage: number | undefined;

  /** Max amplitude as a number */
  maxAmplitude: number | undefined;

  /** Min amplitude as a number */
  minAmplitude: number | undefined;

  /** the y axis ticks */
  yAxisTicks?: number[];
}

export interface YAxisState {}
