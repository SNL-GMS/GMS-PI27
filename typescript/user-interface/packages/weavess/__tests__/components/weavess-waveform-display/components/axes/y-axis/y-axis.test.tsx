/* eslint-disable react/jsx-props-no-spreading */
/* eslint-disable @typescript-eslint/no-magic-numbers */
import { render } from '@testing-library/react';
import * as React from 'react';

import type { YAxisProps } from '../../../../../../src/ts/components/weavess-waveform-display/components/axes/y-axis/types';
import { YAxis } from '../../../../../../src/ts/components/weavess-waveform-display/components/axes/y-axis/y-axis';

const props: YAxisProps = {
  heightInPercentage: 100,
  minAmplitude: 0,
  maxAmplitude: 100,
  yAxisTicks: [0, 2, 4, 6, 8, 10]
};

describe('Weavess Y Axis', () => {
  it('to be defined', () => {
    expect(YAxis).toBeDefined();
  });

  it('renders', () => {
    const { container } = render(<YAxis {...props} />);
    expect(container).toMatchSnapshot();
  });

  it('calls display', () => {
    const yAxis = new YAxis(props);
    const spy = jest.spyOn(yAxis, 'display');
    yAxis.display();
    expect(spy).toHaveBeenCalledTimes(1);
  });

  it('try to display a negative with no ticks', () => {
    const { container } = render(
      <YAxis {...props} yAxisTicks={undefined} minAmplitude={-0.05} maxAmplitude={0} />
    );
    expect(container).toMatchSnapshot();
  });

  it('try to display a large range without ticks', () => {
    const { container } = render(<YAxis {...props} yAxisTicks={undefined} />);
    expect(container).toMatchSnapshot();
  });

  it('updates when display is called', () => {
    const result = render(<YAxis {...props} />);
    expect(result.container).toMatchSnapshot();

    result.rerender(<YAxis {...props} maxAmplitude={10} />);
    expect(result.container).toMatchSnapshot();
  });
});
