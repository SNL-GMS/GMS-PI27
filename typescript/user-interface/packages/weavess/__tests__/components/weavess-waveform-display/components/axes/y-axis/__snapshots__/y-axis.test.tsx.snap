// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Weavess Y Axis renders 1`] = `
<div>
  <div
    class="y-axis"
    data-cy="y-axis"
    data-max="100"
    data-min="0"
    style="height: calc(100% - 4px); margin: 2px 0px 2px 0px;"
  >
    <svg
      height="100%"
      width="50"
    >
      <g
        class="y-axis__ordinate"
        height="100%"
        transform="translate(49,0)"
      />
    </svg>
  </div>
</div>
`;

exports[`Weavess Y Axis try to display a large range without ticks 1`] = `
<div>
  <div
    class="y-axis"
    data-cy="y-axis"
    data-max="100"
    data-min="0"
    style="height: calc(100% - 4px); margin: 2px 0px 2px 0px;"
  >
    <svg
      height="100%"
      width="50"
    >
      <g
        class="y-axis__ordinate"
        height="100%"
        transform="translate(49,0)"
      />
    </svg>
  </div>
</div>
`;

exports[`Weavess Y Axis try to display a negative with no ticks 1`] = `
<div>
  <div
    class="y-axis"
    data-cy="y-axis"
    data-max="0"
    data-min="-0.05"
    style="height: calc(100% - 4px); margin: 2px 0px 2px 0px;"
  >
    <svg
      height="100%"
      width="50"
    >
      <g
        class="y-axis__ordinate"
        height="100%"
        transform="translate(49,0)"
      />
    </svg>
  </div>
</div>
`;

exports[`Weavess Y Axis updates when display is called 1`] = `
<div>
  <div
    class="y-axis"
    data-cy="y-axis"
    data-max="100"
    data-min="0"
    style="height: calc(100% - 4px); margin: 2px 0px 2px 0px;"
  >
    <svg
      height="100%"
      width="50"
    >
      <g
        class="y-axis__ordinate"
        height="100%"
        transform="translate(49,0)"
      />
    </svg>
  </div>
</div>
`;

exports[`Weavess Y Axis updates when display is called 2`] = `
<div>
  <div
    class="y-axis"
    data-cy="y-axis"
    data-max="10"
    data-min="0"
    style="height: calc(100% - 4px); margin: 2px 0px 2px 0px;"
  >
    <svg
      height="100%"
      width="50"
    >
      <g
        class="y-axis__ordinate"
        height="100%"
        transform="translate(49,0)"
      />
    </svg>
  </div>
</div>
`;
